src/gsy_e/gsy_e_core/rq_job_handler.py

```diff
@@ -182,28 +191,38 @@ def _handle_scm_past_slots_simulation_run(
     aggregator_device_mapping: Dict, saved_state: Dict, job_id: str,
     scenario_name: str, slot_length_realtime: Optional[duration],
     kwargs: Dict
-):
+) -> Optional[Dict]:
     # pylint: disable=too-many-arguments
     """
     Run an extra simulation before running a CN, in case the scm_past_slots parameter is set.
     Used to pre-populate simulation results from past market slots before starting the CN.
     """
     scm_past_slots = saved_state.pop("scm_past_slots", False)
-    if GlobalConfig.IS_CANARY_NETWORK and scm_past_slots:
-        config = _create_config_settings_object(
-            scenario, settings, aggregator_device_mapping)
-        config.end_date = now(tz=pendulum.UTC).subtract(
-            days=gsy_e.constants.SCM_CN_DAYS_OF_DELAY)
-        config.sim_duration = config.end_date - config.start_date
-        GlobalConfig.sim_duration = config.sim_duration
-        GlobalConfig.IS_CANARY_NETWORK = False
-        gsy_e.constants.RUN_IN_REALTIME = False
-        run_simulation(setup_module_name=scenario_name,
-                       simulation_config=config,
-                       simulation_events=events,
-                       redis_job_id=job_id,
-                       saved_sim_state=saved_state,
-                       slot_length_realtime=slot_length_realtime,
-                       kwargs=kwargs)
-        GlobalConfig.IS_CANARY_NETWORK = True
-        gsy_e.constants.RUN_IN_REALTIME = True
+    if not (GlobalConfig.IS_CANARY_NETWORK and scm_past_slots):
+        return None
+
+    # Deepcopy the scenario and settings objects, because they are mutated by the
+    # run_simulation function, and we need the original versions for the subsequent
+    # Canary Network run.
+    scenario_copy = deepcopy(scenario)
+    settings_copy = deepcopy(settings)
+
+    config = _create_config_settings_object(
+        scenario_copy, settings_copy, aggregator_device_mapping)
+    config.end_date = now(tz=gsy_e.constants.TIME_ZONE).subtract(
+        days=gsy_e.constants.SCM_CN_DAYS_OF_DELAY).add(hours=4)
```

Comment:
The `add()` is probably a remnant of testing, right?

Reply:
No, this is actually necessary in order to compensate for the runtime of the simulation. Think of the sequence of events that way:
- gsy-web triggers an SCM past slot simulation with a start time 4 months in the past
- gsy-e receives the event and starts a simulation with a start time 4 months ago, and a duration of `now() - 4 months`. Let's say that this simulation will have a runtime of 1 hour.
- once the simulation finishes, gsy-e then starts a Canary Network, at the current now() which is the now() at the start of the simulation in step #2 + 1 hour. Therefore, the simulation and the Canary Network will leave a gap of 1 hour in the results

In order to compensate for that gap, we need to run the simulation in step #2 for longer than the expected duration at the start of it. I decided to set it to 4 hours, in order to take into account any simulations that might take a long time to calculate their results.

Reply:
Ok, thank you for your explanation. Makes sense.
I was actually confused by the subtraction of SCM_CN_DAYS_OF_DELAY. So if the user starts a CN with start time of 4 months in the past, additional 3 days are subtracted from the start time and the end time. Why is this again? AFAIU it is not a problem, because the start of the actual CN (not the pre-calculation sim) is also 3 days earlier. Right?

Reply:
Exactly. This variable controls the constant delay of the CN compared to real time. Therefore the Canary Network will always start at the current time - SCM_CN_DAYS_OF_DELAY, and  the simulation that runs before it has to run until that time too.

Reply:
Can you please remind me of the reason for this please? Is this in order to have results showing up on the UI when the CN is running?

For this PR I would propose an inline comment, that explains the `add(hours=4)` for the future me to not ask again ;-) Thank a lot!

Reply:
I think that the comment I added will clarify this. The SCM_CN_DAYS_OF_DELAY is there in order to account for delays in sending of measurements from the assets. At the moment, Datadis can send the updated measurements even with days of delay, therefore with such a delay we can be certain that we will have received the correct and updated measurement data from all assets.